### PR TITLE
NodeJS plugin

### PR DESCRIPTION
**Pull request by sergiusens**

### CODE REVIEW

File: snapcraft/plugins/nodejs.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import os
+import platform
+
+import snapcraft
+from snapcraft import sources
+
+logger = logging.getLogger(__name__)
+
+_NODEJS_BASE = 'node-v{version}-linux-{arch}'
+_NODEJS_VERSION = '4.2.2'
```
Comment (ted-gould) [inline snapcraft/plugins/nodejs.py]: Seems like this should be a configuration option, no?

Comment (sergiusens) [inline snapcraft/plugins/nodejs.py]: Yeah, was my plan to do this in the future, as in different PR, this is already too big as it is. Good thing is 4.2.2 is an LTS

File: snapcraft/plugins/nodejs.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+"""The nodejs plugin is useful for node/npm based parts.
+
+The plugin uses node to install dependencies from `package.json`. It
+also sets up binaries defined in `package.json` into the `PATH`.
+
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+
+Additionally, this plugin uses the following plugin specific keywords:
```
Comment (chipaca) [inline snapcraft/plugins/nodejs.py]: “plugin-specific” (i.e., add a hyphen)

**Comment by chipaca** [comment]
:+1:

File: snapcraft/plugins/nodejs.py (hunk 2 above)
Comment (sergiusens) [inline snapcraft/plugins/nodejs.py]: I'll make a global change for all the plugins with this.

**Pull request merged**